ibis/sql/udf.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2015 Cloudera Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hashlib
+
+from ibis.expr.types import (
+    Int8Value, Int16Value,
+    Int32Value, Int64Value,
+    BooleanValue, FloatValue,
+    DoubleValue, StringValue,
+    TimestampValue, DecimalValue,
+)
+from ibis.common import IbisTypeError
+
+import ibis.expr.types as ir
+import ibis.expr.operations as _ops
+import ibis.expr.rules as rules
+import ibis.sql.exprs as _expr
+import ibis.util as util
+
+
+class UDFInfo(object):
+
+    def __init__(self, input_type, output_type, name):
+        self.inputs = input_type
+        self.output = output_type
+        self.name = name
+
+
+class UDFCreatorParent(UDFInfo):
+
+    def __init__(self, hdfs_file, input_type,
+                 output_type, name=None):
+        if hdfs_file[-3:] != '.so':
+            raise ValueError('File is not a .so file')
+        self.hdfs_file = hdfs_file
+        inputs = [ir._validate_type(x) for x in input_type]
+        output = ir._validate_type(output_type)
+        new_name = name
+        if not name:
+            new_name = hashlib.sha1(self.so_symbol).hexdigest()
```

Comment:
Can you join `so_symbol` with `hdfs_path` to reduce chances for collisions (e.g. two .so files containing the same symbol name -- the sha1 hash would be the same)?